src/fsharp/ReferenceResolution.fs

```diff
@@ -106,7 +112,10 @@ module internal MSBuildResolver =
             | Net35 ->  Some TargetDotNetFrameworkVersion.Version35
             | Net40 ->  Some TargetDotNetFrameworkVersion.Version40
             | Net45 ->  Some TargetDotNetFrameworkVersion.Version45
+#if BUILDING_PROTO
+#else
             | Net451 -> Some TargetDotNetFrameworkVersion.Version451
```

Comment:
We should not need to make these changes... `Net451` is a valid enum value as of msbuild 12.0 (docs [here](https://msdn.microsoft.com/en-us/library/ms126273(v=vs.121).aspx)).  I believe this is only causing a problem because the proto msbuild version was changes down to 4.0 from 12.0.